cmake/modules/ALL.cmake

```diff
@@ -1,5 +1,10 @@
 # cmake module for adding ALL
 
+option(ENABLE_GPROF "Use the GNU profiler (gprof)" OFF)
+if(ENABLE_GPROF)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pg")
+endif()
```

Comment:
- [ ] Why is this in the module for `ALL`? IMHO this should go either into the ominous `options.cmake` or a new module.

- [ ] Add to the doc string that this is not supported by all compilers. AFAIK only gcc and clang support this but I wouldn't artificially limit this option only to those. If you want to be really tidy throw a cmake warning if this option is used with any other compiler.